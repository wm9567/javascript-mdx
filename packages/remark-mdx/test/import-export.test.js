const extract = require('../extract-imports-and-exports')
const fixtures = require('./fixtures/import-export')

fixtures.forEach(fixture => {
  it(fixture.description, () => {
    const result = extract(fixture.mdx)

    expect(result).toMatchSnapshot(fixture.result)
  })
})
