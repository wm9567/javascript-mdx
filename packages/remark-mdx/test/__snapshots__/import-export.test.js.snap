// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handles an import that is later exported 1`] = `
Array [
  Object {
    "type": "import",
    "value": "import Layout from './Foo'

",
  },
  Object {
    "default": true,
    "type": "export",
    "value": "export default Layout",
  },
]
`;

exports[`Handles basic usecase of default export 1`] = `
Array [
  Object {
    "default": true,
    "type": "export",
    "value": "export default props => <article {...props} />",
  },
]
`;

exports[`Handles const exports 1`] = `
Array [
  Object {
    "type": "export",
    "value": "export const metadata = { some: \\"stuff\\" }",
  },
]
`;

exports[`Handles multiline default exports 1`] = `
Array [
  Object {
    "default": true,
    "type": "export",
    "value": "export default props => (
  <main {...props} />
)",
  },
]
`;

exports[`Handles multiline exports 1`] = `
Array [
  Object {
    "type": "export",
    "value": "export const metadata = {
  some: \\"stuff\\"
}",
  },
]
`;

exports[`Separates import from the default export 1`] = `
Array [
  Object {
    "type": "import",
    "value": "import Foo from './foo'
",
  },
  Object {
    "default": true,
    "type": "export",
    "value": "export default props => <article {...props} />",
  },
]
`;
